src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor_Constants.cs

```diff
@@ -1,59 +1,40 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 using System.Diagnostics;
-using System.Reflection;
 
 namespace Microsoft.CodeAnalysis.CSharp
 {
     internal partial class SymbolDisplayVisitor
     {
-        private void AddConstantValue(ITypeSymbol type, object constantValue, bool preferNumericValueOrExpandedFlagsForEnum = false)
+        protected override void AddExplicitlyCastedLiteralValue(INamedTypeSymbol namedType, object value)
         {
-            if (constantValue != null)
-            {
-                AddNonNullConstantValue(type, constantValue, preferNumericValueOrExpandedFlagsForEnum);
-            }
-            else if (type.IsReferenceType || type.TypeKind == TypeKind.Pointer || ITypeSymbolHelpers.IsNullableType(type))
-            {
-                AddKeyword(SyntaxKind.NullKeyword);
-            }
-            else
-            {
-                AddKeyword(SyntaxKind.DefaultKeyword);
-                AddPunctuation(SyntaxKind.OpenParenToken);
-                type.Accept(this.NotFirstVisitor);
-                AddPunctuation(SyntaxKind.CloseParenToken);
-            }
-        }
+            Debug.Assert(namedType.TypeKind == TypeKind.Enum);
 
-        protected override void AddExplicitlyCastedLiteralValue(INamedTypeSymbol namedType, SpecialType type, object value)
-        {
             AddPunctuation(SyntaxKind.OpenParenToken);
             namedType.Accept(this.NotFirstVisitor);
             AddPunctuation(SyntaxKind.CloseParenToken);
-            AddLiteralValue(type, value);
+            AddNonEnumConstantValue(namedType.EnumUnderlyingType, value);
         }
 
-        protected override void AddLiteralValue(SpecialType type, object value)
+        private static bool CanDefinitelyBeNull(ITypeSymbol type)
```

Comment:
Maybe 'CanUseNullLiteral'?

Reply:
Can definitely be used... :smile: my idea with "definitely" was to make it clear that this might not handle all cases - it doesn't return true for constrained type parameters. The important part is that when it does, we can use null.

Reply:
Gotcha :)  i'm fine either way.  Thanks!

Reply:
➡️ Renamed in 5e47bdaf56331c8ea0214ad9afe0f3848fc255f0